tracer/src/Datadog.Trace/Iast/Propagation/PropagationModuleImpl.cs

```diff
@@ -15,6 +15,39 @@ internal static class PropagationModuleImpl
 {
     private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(PropagationModuleImpl));
 
+    public static void AddTainted(string? input, Source source)
+    {
+        try
+        {
+            if (string.IsNullOrEmpty(input))
+            {
+                return;
+            }
+
+            var iastContext = IastModule.GetIastContext();
+            if (iastContext is null)
+            {
+                return;
+            }
+
+            var taintedObjects = iastContext.GetTaintedObjects();
+#pragma warning disable CS8604 // Possible null reference argument.
```

Comment:
could you use a cast or pattern matching on `input` to avoid these pragmas?

Reply:
I could, for instance, do a "input != null" check to avoid these ugly pragmas. I hope that in the future the compiler will be improved and it will realize that string.IsNullOrEmpty already checks for null values. I will remove the pragmas.